include/aktualizr-lite/tuf/tuf.h

```diff
@@ -0,0 +1,142 @@
+#ifndef AKLITE_TUF_TUF_H_
+#define AKLITE_TUF_TUF_H_
+
+#include <iostream>
+#include <string>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <boost/program_options/variables_map.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include "crypto/keymanager.h"
+#include "http/httpclient.h"
+#include "json/json.h"
+#include "libaktualizr/config.h"
+#include "storage/invstorage.h"
+#include "uptane/fetcher.h"
+#include "uptane/imagerepository.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+
+namespace Uptane {
+class Target;
+}
+
+namespace aklite {
+namespace tuf {
+
+// enum class RoleEnum { Root = 0, Snapshot = 1, Targets = 2, Timestamp = 3, InvalidRole = -1 };
+// virtual std::string fetchRole(RoleEnum role, int version) = 0;
+
+// TODO: consider switching to a single fetchRole
+class TufRepoSource {
+ public:
+  virtual std::string fetchRoot(int version) = 0;
+  virtual std::string fetchTimestamp() = 0;
+  virtual std::string fetchSnapshot() = 0;
+  virtual std::string fetchTargets() = 0;
+};
+
+// TufRepoSource implementation for fetching remote meta-data via https using libaktualizr
+class AkHttpsTufRepoSource : public TufRepoSource {
+ public:
+  AkHttpsTufRepoSource(std::string name_in, boost::property_tree::ptree& pt);
+  ~AkHttpsTufRepoSource();
+
+  virtual std::string fetchRoot(int version);
+  virtual std::string fetchTimestamp();
+  virtual std::string fetchSnapshot();
+  virtual std::string fetchTargets();
+
+ private:
+  void init(std::string name_in, boost::property_tree::ptree& pt);
+  void fillConfig(Config& config, boost::property_tree::ptree& pt);
+  std::string fetchRole(Uptane::Role role, int64_t maxsize, Uptane::Version version);
+
+  std::string name_;
+  boost::filesystem::path tmp_dir_path_;
+  std::shared_ptr<Uptane::IMetadataFetcher> meta_fetcher_;
+};
+
+// TufRepoSource implementation for fetching local meta-dat
+class AkLocalTufRepoSource : public TufRepoSource {
+ public:
+  class NotFoundException : public std::runtime_error {
+   public:
+    NotFoundException(const std::string& path) : std::runtime_error("Metadata hasn't been found; file path: " + path) {}
+  };
+
+  AkLocalTufRepoSource(std::string name_in, boost::property_tree::ptree pt);
```

Comment:
Why do we need to ctors? And why do we need `boost::property_tree::ptree` to initialize the local repo source?

Reply:
Not needed, but it made the constructor consistent with the one for AkHttpsTufRepoSource, which requires a few more settings than just the uri. I'll keep just the one receiving the uri, and drop the one receiving the ptree.

Reply:
Done